### PR TITLE
Updated code to set the text encoding (ccsid) value

### PR DESCRIPTION
**Pull request by milesje**
My only concern with this pull request is that it defaults the ccsid value to 1252 (Windows, Latin1) encoding and this may not be the preferred default.

### CODE REVIEW

**Comment by milesje** [comment]
Also of concern is that the test fail. Although the test did not work before any of my modification either.

**Comment by cwg999** [comment]
> My only concern with this pull request is that it defaults the ccsid value to 1252 (Windows, Latin1) encoding and this may not be the preferred default.

Rather than provide a default, simply leave it as undefined?

**Comment by jakobrun** [comment]
Thanks for the contribution @milesje. The default ccsid value also concerns me. One solution would be to use Integer and if it is null then use the constructor without ccsid and default to null. Also I don't understand why all the jar files have changed. jt400wrap.jar is the only jar file that should change.

**Comment by milesje** [comment]
I will work on changing it so that the ccsid is set via a parameter instead of the options object. This way it can be check for null and then use the constructor without the ccsid.
I'm not sure why all the jar files have changed; I didn't do anything to the jar files. I made my code changes and performed `npm run build-all`

**Comment by cwg999** [comment]
Just do a 

`git checkout origin/master -- java/lib/jt400.jar` etc. for all of those files (except jt400wrap.jar)

I'm not sure why it happened to me either. Might be windows doing something funky first time it's accessed?

File: lib/jt400.ts [hunk 1]
```diff
@@ -349,7 +349,7 @@ export interface IfsFileMetadata {
 
 export interface Ifs {
 	createReadStream: (fileName: string | Promise<string>) => Readable
-	createWriteStream: (fileName: string | Promise<string>, options?: { append: boolean }) => Writable
+	createWriteStream: (fileName: string | Promise<string>, options?: { append: boolean, ccsid: 1252 }) => Writable
```
Comment (bergur) [inline lib/jt400.ts]: This is the type interface and should be a number, not 1252

**Comment by bergur** [comment]
Hi @milesje, this is now on the master. 

There were some conflicts that I had to resolve locally. I merged your code into the master with one slight change: I made the ccsid property on the options object optional, so that other users wouldn't get type errors when the only option is append.

Thank you for your help and patience.

**Comment by bergur** [comment]
Hi @milesje, this is now on the master.

There were some conflicts that I had to resolve locally. I merged your code into the master with one slight change: I made the ccsid property on the options object optional, so that other users wouldn't get type errors when the only option is append. Other than that it worked like a charm.

Thank you for your help and patience.

**Comment by bergur** [comment]
If anyone knows how I can mark this as merged, then that would be appreciated. I just followed github instructions but I might have screwed up.

**Comment by cwg999** [comment]
I feel like you'd have to merge in his commit for it to mark it as merged... unfortunately.

**Comment by bergur** [comment]
Hmm I believe I did. I pulled, fixed the conficts, merged and pushed, as instructed by the github instructions.

It looks like a general problem:

"Pull requests are automatically marked as merged when the commits are not modified before pulling them into the main branch.

However, for repos that receive a lot of pull requests, the option is to either (a) have an excessive number of merge commits, or (b) force pull requestors to continually rebase on the lastest upstream, or (c) rebase/cherry-pick/am the patches on my repo, and forego the "merged" data point

Due to the impossibility of automatically determining whether a pull request was merged, it looks like Node doesn't accept patches.

It'd be great if I could explicitly indicate in the UI that a pull request was merged, so that the data could be more accurate."

I will look better into this tomorrow.

**Comment by cwg999** [comment]
I saw that exact forum post, my digestion of it was that option (a) means merge his commit into master at this point, fixing the conflicts and re-doing what you've done in your (different hash) commit <yuck, lots of garbage commits>, (b) means that he would had to rebase so that you could make the modifications in a clean, single commit <good luck with time/phase lag>, or (c) means yes, lets ignore the nice merge tag and just bring them in. <what you did?> I'm no git expert.

**Pull request closed without merging**